### PR TITLE
feat: added more primitives

### PR DESCRIPTION
**Pull request by kyranet**
Added `s.bigint`
Added `s.boolean`
Added `s.date`
Added `s.literal(Date)`
Added `s.never`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -134,6 +134,63 @@ s.number.negative; // .lt(0)
...
+
+ShapeShift includes a handful of number-specific validations:
+
+```typescript
+s.bigint.gt(5); // > 5
+s.bigint.ge(5); // >= 5
+s.bigint.lt(5); // < 5
+s.bigint.le(5); // <= 5
+s.bigint.eq(5); // === 5
+s.bigint.ne(5); // !== 5
```
Comment (vladfrangu) [inline README.md:160]:
```suggestion
s.bigint.gt(5); // > 5n
s.bigint.ge(5); // >= 5n
s.bigint.lt(5); // < 5n
s.bigint.le(5); // <= 5n
s.bigint.eq(5); // === 5n
s.bigint.ne(5); // !== 5n
```

File: README.md [hunk 2]
```diff
@@ -134,6 +134,63 @@ s.number.negative; // .lt(0)
...
+
+s.bigint.finite;  // value must be finite
+
+s.bigint.positive; // .ge(0)
+s.bigint.negative; // .lt(0)
```
Comment (vladfrangu) [inline README.md:165]:
```suggestion
s.bigint.positive; // .ge(0n)
s.bigint.negative; // .lt(0n)
```

File: README.md [hunk 3]
```diff
@@ -134,6 +134,63 @@ s.number.negative; // .lt(0)
...
+
+s.bigint.divisibleBy(5); // TODO | Divisible by 5
```
Comment (vladfrangu) [inline README.md:167]:
```suggestion
s.bigint.divisibleBy(5n); // TODO | Divisible by 5n
```

File: README.md [hunk 4]
```diff
@@ -134,6 +134,63 @@ s.number.negative; // .lt(0)
...
+```
+
+And transformations:
+
+```typescript
+s.number.abs;  // TODO | Transforms the bigint to an absolute bigint
```
Comment (vladfrangu) [inline README.md:173]: Bigints are absolute by definition?

File: README.md [hunk 5]
```diff
@@ -134,6 +134,63 @@ s.number.negative; // .lt(0)
...
+
+s.bigint.intN(5);  // TODO | Clamps to a bigint to a signed bigint with 5 digits, see BigInt.asIntN
+s.bigint.uintN(5); // TODO | Clamps to a bigint to an unsigned bigint with 5 digits, see BigInt.asUintN
```
Comment (vladfrangu) [inline README.md:176]:
```suggestion
s.bigint.intN(5n);  // TODO | Clamps to a bigint to a signed bigint with 5 digits, see BigInt.asIntN
s.bigint.uintN(5n); // TODO | Clamps to a bigint to an unsigned bigint with 5 digits, see BigInt.asUintN
```

**Review by vladfrangu — CHANGES_REQUESTED** [review]

File: README.md (hunk 4 above)
Comment (kyranet) [reply README.md:173]: No, `-1n`

File: README.md (hunk 5 above)
Comment (kyranet) [reply README.md:176]: Actually, those take numbers, see [`BigInt.asIntN()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt/asIntN) and [`BigInt.asUintN()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt/asUintN)

Comment (vladfrangu) [reply README.md:176]: It shouldn't take number, and I'm dying on that hill. This is a bigint validator, not a numberesque one

Comment (kyranet) [reply README.md:176]: Still, that method takes a number.

File: README.md [hunk 6]
```diff
@@ -134,6 +134,63 @@ s.number.negative; // .lt(0)
...
+
+ShapeShift includes a handful of number-specific validations:
+
+```typescript
+s.bigint.gt(5); // > 5n
+s.bigint.ge(5); // >= 5n
+s.bigint.lt(5); // < 5n
+s.bigint.le(5); // <= 5n
+s.bigint.eq(5); // === 5n
+s.bigint.ne(5); // !== 5n
```
Comment (vladfrangu) [inline README.md:160]:
```suggestion
s.bigint.gt(5n); // > 5n
s.bigint.ge(5n); // >= 5n
s.bigint.lt(5n); // < 5n
s.bigint.le(5n); // <= 5n
s.bigint.eq(5n); // === 5n
s.bigint.ne(5n); // !== 5n
```

File: README.md [hunk 7]
```diff
@@ -134,6 +134,63 @@ s.number.negative; // .lt(0)
...
+
+s.bigint.finite;  // value must be finite
```
Comment (vladfrangu) [inline README.md:162]: Aren't bigints by definition..finite?

File: README.md (hunk 4 above)
Comment (vladfrangu) [reply README.md:173]:
```suggestion
s.bigint.abs;  // TODO | Transforms the bigint to an absolute bigint
```

**Review by vladfrangu — CHANGES_REQUESTED** [review]

**Review by vladfrangu — APPROVED** [review]

**Pull request merged**